File: server/service/leave.go
package service

type LeaveService interface {
	AddLeave(AddLeaveRequest, uint) (*LeaveResponse, error)
	GetLeavesByUser(uint) (*LeaveResponseWithCount, error)
	GetLeaves() ([]LeaveResponse, error)
}

type AddLeaveRequest struct {
	Type      string `json:"type"`
	Detail    string `json:"detail"`
	TimeStart string `json:"time_start"`
	TimeEnd   string `json:"time_end"`
}

type LeaveResponse struct {
	ID        uint   `json:"id"`
	Type      string `json:"type"`
	Detail    string `json:"detail"`
	TimeStart string `json:"time_start"`
	TimeEnd   string `json:"time_end"`
	CreatedAt string `json:"created_at"`
	UpdatedAt string `json:"updated_at"`
	LeaveDay  uint   `json:"leave_day"`
}

type LeaveResponseWithCount struct {
	LeaveResponses []LeaveResponse `json:"leave_responses"`
	CountSick      uint            `json:"count_sick"`
	CountBusiness  uint            `json:"count_business"`
	CountVacation  uint            `json:"count_vacation"`
}

File: client/store/leave.ts
import { createSlice } from "@reduxjs/toolkit";


export interface LeaveDTO {
    id: number
    type: string
    time_start: string
    time_end: string
    detail: string
    created_at: string
    updated_at: string
    leave_day: number
}
export interface LeaveReponseWithCountDTO {
    leave: LeaveDTO[]
    count_sick: number
    count_business: number
    count_vacation: number
}

const leaveSlice = createSlice({
    name: "leave",
    initialState: {
        leave: [] as LeaveDTO[] | null,
    },
    reducers: {
        setLeave: (state, action) => {
            state.leave = action.payload;
        },
        addLeave: (state, action) => {
            return
        }
    },
});

export const { setLeave, addLeave } = leaveSlice.actions;
export default leaveSlice.reducer;
